code/entities/chair/CinemaChair.cs

```diff
@@ -0,0 +1,214 @@
+    /// is currently seated here.
+    /// </summary>
+    [Net]
+    public Player Occupant { get; protected set; }
+
+    /// <summary>
+    /// Allows access to both of the armrests associated with this chair.
+    /// </summary>
+    [Net]
+    public IDictionary<Armrest.Sides, Armrest> Armrests { get; set; }
+
+    public Armrest LeftArmrest
+        => Armrests[Armrest.Sides.Left];
+    public Armrest RightArmrest
+        => Armrests[Armrest.Sides.Right];
+
+    /// <summary>
+    /// An offset from the origin of this chair that shall be used to determine the position 
+    /// of an entity sitting in this chair.
+    /// </summary>
+    [Net, Property]
+    public Vector3 SeatOffset { get; set; } = new Vector3(0, 0, 0);
+
+    /// <summary>
+    /// An offset from the origin of this chair that shall be used to determine the position
+    /// at which an entity will be placed after it has stopped sitting in this chair.
+    /// </summary>
+    [Net, Property]
+    public Vector3 EjectOffset { get; set; }
+
+    /// <summary>
+    /// Specifies the angles from which this chair is usable. Based on
+    /// the dot product of the chair's forward direction and the direction 
+    /// vector between the player and the chair. For example,
+    /// 
+    ///     -1:     Usable from any direction
+    ///     0:      Usable from the size or front
+    ///     0.5:    Usable from the front
+    /// </summary>
+    [Net, Property]
+    public float UseAngle { get; set; } = 0.7f;
+
+    [Net, Property]
+    public Vector3 EyeOffset { get; set; } = new Vector3(0, 0, 64);
+
+    public string UseText => "Sit Down";
+    public string CannotUseText
+    {
+        get
+        {
+            if (IsOccupied)
+            {
+                return "Seat is occupied.";
+            }
+            else
+            {
+                return null;
+            }
+        }
+    }
+
+    public override void Spawn()
+    {
+        base.Spawn();
+
+        Tags.Add("solid", "chair");
+
+        var capsule = new Capsule(
+            Vector3.Zero,
+            new Vector3(0, 0, 64),
+            16f
+            );
+        SetupPhysicsFromCapsule(PhysicsMotionType.Keyframed, capsule);
+
+        Armrests[Armrest.Sides.Left] = new Armrest()
+        {
+            Chair = this,
+            Side = Armrest.Sides.Left
+        };
+        Armrests[Armrest.Sides.Right] = new Armrest()
+        {
+            Chair = this,
+            Side = Armrest.Sides.Right
+        };
+    }
+
+    public bool IsUsable(Entity user)
+    {
+        if (!user.IsValid)
+        {
+            return false;
+        }
+        var direction = (user.Position - Position).Normal;
+        var angle = direction.Dot(Rotation.Forward);
+        Log.Trace($"Chair use angle: {angle}, Min valid angle: {UseAngle}");
+        // Check whether the player is in front of the seat.
+        if (angle < UseAngle)
+        {
+            return false;
+        }
+        return !IsOccupied;
```

Comment:
Performance best practice would be to check this first and return false if it's occupied.

Then, check the angle. Reason being is that `.Normal` requires a square root operation which is expensive.